pages/api/openPR.js

```diff
@@ -15,40 +15,199 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
+        organizations: values.organizations ? [values.organizations] : [],
+        stage: values.stage ? values.stage : "",
+      })
+  );
+
+  // Order nominee fields in the correct order e.g organizations
+  sortedSubmission = orderFields(sortedSubmission);
+  return sortedSubmission;
+}
+
+// DPG candidate processing before opening pull request
+function dpgSubmission(values, sortedSubmission) {
+  // Sort entries by iterating through object and unpacking entries
+  Object.entries(values).forEach(
+    ([key, value]) =>
+      (sortedSubmission = {
+        name: values.name ? values.name : "",
+        aliases: values.aliases ? [values.aliases] : [""],
+        description: values.description ? values.description : "",
+        website: values.website ? values.website : "",
+        license: values.license ? values.license : [],
+        SDGs: values.SDGs ? values.SDGs : [],
+        sectors: values.sectors ? values.sectors : [],
+        type: values.type ? values.type : [],
+        repositoryURL: values.repositoryURL ? values.repositoryURL : "",
+        organizations: values.organizations ? [values.organizations] : [],
+        stage: values.stage ? values.stage : "",
+      })
+  );
+
+  // Order fields in the correct order e.g nominee organizations
+  sortedSubmission = orderFields(sortedSubmission);
+  return sortedSubmission;
+}
+
 export default async (req, res) => {
   if (req.method === "POST") {
-    const values = req.body.values;
-    console.log(values);
+    let values = req.body.values;
+
+    let myJSON, sdgNumber, evidenceText, sortedSubmission;
 
     // Exclude contact information from pull request
     delete values["contact"];
 
-    let sdgNumber, evidenceText;
-    for (let i = 0; i < values.SDGs.length; i++) {
-      sdgNumber = parseInt(values.SDGs[i]);
-      evidenceText = "evidenceText".concat(sdgNumber);
-
-      values.SDGs[i] = {
-        SDGNumber: sdgNumber,
-        evidenceText: values[evidenceText],
-      };
-
-      delete values[evidenceText];
-    }
-
-    // Convert JavaScript object to JSON string
-    let myJSON = JSON.stringify(values, null, "\t");
+    // Parse SDG information
+    values = getSDGRelevanceInfo(values, sdgNumber, evidenceText);
 
-    // Add newline at end of file
-    myJSON += "\r\n";
+    // Verify submission stage(nominee/ DPG) and channel to nomination or DPG review processing
+    sortedSubmission =
+      values.stage === "nominee"
+        ? nomineeSubmission(values, sortedSubmission)
+        : dpgSubmission(values, sortedSubmission);
 
-    // Return project json file aligned with naming convention (includes removing accents)
-    let name =
-      values.name
-        .normalize("NFD")
-        .toLowerCase()
-        .replace(/[\u0300-\u036f]/g, "")
-        .replace(/ /g, "-") + ".json";
+    // Convert JavaScript submission sorted object into JSON string and add newline at EOF
+    myJSON = JSON.stringify(sortedSubmission, null, 2).concat("\n");
```

Comment:
Please use more descriptive variables, instead of `myJSON` consider `nomineeJSON`

Reply:
Resolved in commit b645ff6.